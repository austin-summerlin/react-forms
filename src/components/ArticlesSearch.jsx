import React from 'react';
import PropTypes from 'prop-types';

const ArticlesSearch = ({ search, onChange, onSubmit }) => (
  <>
    <form onSubmit={onSubmit}>
      <label htmlFor="search">Search</label>
      <input
        id="search"
        type="search"
        value={search}
        onChange={onChange}
        placeholder="Search"
      />
      <button aria-label="button">Search</button>
    </form>
  </>
);

ArticlesSearch.propTypes = {
  search: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default ArticlesSearch;
