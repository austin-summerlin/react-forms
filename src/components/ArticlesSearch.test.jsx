/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import ArticlesSearch from './ArticlesSearch';

describe('Article Search Component', () => {
  it('should render the component', () => {
    const search = 'berlin';
    const { asFragment } = render(<ArticlesSearch ArticlesSearch={search} />);

    expect(asFragment).toMatchSnapshot();
  });
});
