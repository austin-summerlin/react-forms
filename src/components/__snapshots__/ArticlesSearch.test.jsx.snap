// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Search Component should render the component 1`] = `[Function]`;
